modules/libjpeg_turbo/2.1.91/patches/add_build_file.patch

```diff
@@ -0,0 +1,193 @@
++]
++
++SUBSTITUTIONS.update({
++    "#cmakedefine " + key: "#define " + key
++    for key in YES_DEFINES
++})
++
++SUBSTITUTIONS.update({
++    "#cmakedefine " + key: "// #define " + key
++    for key in NO_DEFINES
++})
++
++[
++    expand_template(
++        name = "expand_" + src,
++        out = src,
++        substitutions = SUBSTITUTIONS,
++        template = src + ".in",
++        visibility = ["//visibility:public"],
++    )
++    for src in [
++        "jconfig.h",
++        "jconfigint.h",
++        "jversion.h",
++    ]
++]
++
++JPEG16_SOURCES = [
++    "jccolor.c",
++    "jcdiffct.c",
++    "jclossls.c",
++    "jcmainct.c",
++    "jcprepct.c",
++    "jcsample.c",
++    "jdcolor.c",
++    "jddiffct.c",
++    "jdlossls.c",
++    "jdmainct.c",
++    "jdmerge.c",
++    "jdpostct.c",
++    "jdsample.c",
++    "jquant1.c",
++    "jquant2.c",
++    "jutils.c",
++]
++
++JPEG12_SOURCES = JPEG16_SOURCES + [
++    "jccoefct.c",
++    "jcdctmgr.c",
++    "jdcoefct.c",
++    "jddctmgr.c",
++    "jfdctfst.c",
++    "jfdctint.c",
++    "jidctflt.c",
++    "jidctfst.c",
++    "jidctint.c",
++    "jidctred.c",
++]
++
++JPEG_SOURCES = JPEG12_SOURCES + [
++    "jaricom.c",
++    "jcapimin.c",
++    "jcapistd.c",
++    "jcarith.c",
++    "jchuff.c",
++    "jcicc.c",
++    "jcinit.c",
++    "jclhuff.c",
++    "jcmarker.c",
++    "jcmaster.c",
++    "jcomapi.c",
++    "jcparam.c",
++    "jcphuff.c",
++    "jdapimin.c",
++    "jdapistd.c",
++    "jdarith.c",
++    "jdatadst.c",
++    "jdatasrc.c",
++    "jdhuff.c",
++    "jdicc.c",
++    "jdinput.c",
++    "jdlhuff.c",
++    "jdmarker.c",
++    "jdmaster.c",
++    "jdphuff.c",
++    "jdtrans.c",
++    "jerror.c",
++    "jfdctflt.c",
++    "jmemmgr.c",
++    "jmemnobs.c",
++]
++
++JPEG_HEADERS = [
++    "jccolext.c",
++    "jchuff.h",
++    "jcmaster.h",
++    "jconfig.h",
++    "jconfigint.h",
++    "jdcoefct.h",
++    "jdcol565.c",
++    "jdcolext.c",
++    "jdct.h",
++    "jdhuff.h",
++    "jdmainct.h",
++    "jdmaster.h",
++    "jdmerge.h",
++    "jdmrg565.c",
++    "jdmrgext.c",
++    "jdsample.h",
++    "jerror.h",
++    "jinclude.h",
++    "jlossls.h",
++    "jmemsys.h",
++    "jmorecfg.h",
++    "jpeg_nbits_table.h",
++    "jpegapicomp.h",
++    "jpegint.h",
++    "jpeglib.h",
++    "jsamplecomp.h",
++    "jsimd.h",
++    "jsimddct.h",
++    "jstdhuff.c",
++    "jversion.h",
++]
++
++JPEG_DEFINES = select({
++    "@platforms//os:windows": ["LJT_INLINE=__inline"],
++    "//conditions:default": ["LJT_INLINE=__inline"],
++})
++
++cc_library(
++    name = "jpeg16",
++    srcs = JPEG16_SOURCES,
++    hdrs = JPEG_HEADERS,
++    defines = JPEG_DEFINES,
++    local_defines = ["BITS_IN_JSAMPLE=16"],
++    visibility = ["//visibility:public"],
++)
++
++cc_library(
++    name = "jpeg12",
++    srcs = JPEG12_SOURCES,
++    hdrs = JPEG_HEADERS,
++    defines = JPEG_DEFINES,
++    local_defines = ["BITS_IN_JSAMPLE=12"],
++    visibility = ["//visibility:public"],
++)
++
++cc_library(
++    name = "jpeg",
```

Comment:
This results in a library called `libjpeg.so` on Linux. Is that what we want or should this be called `jpeg_turbo` instead?

In either case, could you add an alias named `libjpeg_turbo` so that users can depend on this module via the `@libjpeg_turbo` shorthand?

Reply:
In native build there are two libraries (lib)`jpeg` and (lib)`turbojpeg`. The former is kind-of normal JPEG library replacement. The latter adds some extras.
Current build file copies behavior of (lib)`jpeg,` so let's stick with `jpeg` library plus alias.

Reply:
In that case, could the alias end up being confusing? I didn't know that the native build creates two libraries - in this case, forcing the user to make a conscious choice may be better.

Reply:
We could drop alias for now and listen to what actual users will ask.
As I perceive BCR - it unburdens from writing missing BUILD files; what to do with resulting artifacts is up to embedder...